Storage/SQL/BrowserDB.swift

```diff
@@ -233,26 +224,26 @@ open class BrowserDB {
 
         let deferred = Success()
 
-        var error: NSError?
-        error = self.transaction(synchronous: false, err: &error) { (conn, err) -> Bool in
+        if let err = self.transaction(synchronous: false, { (conn) -> Bool in
             for (sql, args) in commands {
-                err = conn.executeChange(sql, withArgs: args)
-                if let err = err {
-                    deferred.fill(Maybe(failure: DatabaseError(err: err)))
-                    return false
+                if let err = conn.executeChange(sql, withArgs: args) {
+                    log.warning("SQL operation failed: \(err.localizedDescription)")
+                    throw err
                 }
             }
 
             deferred.fill(Maybe(success: ()))
             return true
+        }) {
+            deferred.fill(Maybe(failure: DatabaseError(err: err)))
```

Comment:
https://github.com/mozilla-mobile/firefox-ios/pull/3197/files?diff=unified#diff-deda0d7cd11ef1ffd067e82f0899bb76R1111 is how we can get an error here after having the inner block succeed.